### ISSUE TITLE
Is it possible to make a HOG face detection with GPU?

### ISSUE THREAD

**Issue by apaticmusic** [ageitgey/face_recognition#610]
I found that CNN face detector found not only the fromtal faces, but for my purpose its too match. I need only frontal faces and i found that HOG method giving better results for me. Is it possible to make in work on GPU like CNN can work with batches?

**Comment by ageitgey** [comment]
No, the HOG detector isn't implemented in a way that would currently benefit from a GPU/CUDA as far as I know. 

I suppose it's possible someone could dig into the dlib HOG/svm code and try to add some GPU speed-ups, but I'm not sure how much work it would take or how much benefit you would even get for that anyway. CUDA is really designed for speeding up the specific math used in neural networks and thus works best for those kinds of algorithms. I'm not familiar enough with CUDA's API to know if it would be a good fit for speeding up anything in the HOG/svm pipeline.

**Comment by adeoy** [comment]
There is anyway to speed up more the face recognition with HOG? I already use all my cores and threads of CPU and it can analyze near 50k images per second (already been recognized and face landmarks stored in a database), but I don't know if it can go more faster than that.

**Comment by 7thstorm** [comment]
any updates of getting HOG to work with GPU?

**Comment by shiva2410** [comment]
@adeoy bro can you share your code . How you managed to run hog based face detector on Multiple cpus I am looking to do same from past few days

**Comment by Hakentha** [comment]
50k per image? share code please.
